src/pybind/mgr/deepsea/module.py

```diff
@@ -24,23 +26,9 @@ def __init__(self, message, status_code=None):
         self.status_code = status_code
 
 
-class DeepSeaReadCompletion(orchestrator.ReadCompletion):
-    def __init__(self, process_result_callback):
-        super(DeepSeaReadCompletion, self).__init__()
-        self._complete = False
-        self._cb = process_result_callback
-
+class DeepSeaReadCompletion(orchestrator.Completion):
     def _process_result(self, data):
-        self._result = self._cb(data)
-        self._complete = True
-
-    @property
-    def result(self):
-        return self._result
-
-    @property
-    def has_result(self):
-        return self._complete
+        self.finalize(data)
```

Comment:
I could be wrong, but I think this change means the two calls to `c = DeepSeaReadCompletion(process_result)` elsewhere in this module need to be changed to `c = DeepSeaReadCompletion(on_complete=process_result)`

Reply:
thanks, will fix this

Reply:
resolved